samples/react/telehealth/src/views/DoctorView.tsx

```diff
@@ -1,36 +1,61 @@
-import React from "react";
+import React, { useState } from "react";
 import {
   UserEntity,
   MessageList,
   MessageInput,
-  MemberList,
+  ChannelList,
   Chat,
+  ChannelEntity,
 } from "@pubnub/react-chat-components";
 
 import { ReactComponent as MedicalIcon } from "../assets/clipboard-medical.svg";
 import { ReactComponent as ArrowUp } from "../assets/arrow-turn-up.svg";
+import memberships from "../data/memberships.json";
+import users from "../data/users.json";
 
 type DoctorViewProps = {
-  patient: UserEntity;
   doctor: UserEntity;
-  channel: string;
 };
 
 function DoctorView(props: DoctorViewProps): JSX.Element {
-  const { patient, doctor, channel } = props;
+  const { doctor } = props;
+  const doctorMemberships = memberships.filter((m) => m.members.includes(doctor.id));
+  const channels = doctorMemberships.map((m) => {
```

Comment:
I don't like it at all and I don't like TS casting here.

Since you perform `if (!patient) return;` then the map functions will return something like this [ChannelEntity, ChannelEntity, undefined, undefined, ChannelEntity] which is a mixed-type array and definitely not a ChannelEntity[]. Moreover I once read that mixed-typed arrays are costly for JS engines but it's probably just a curiosity here since dataset is small :P.

Instead I suggest to filter out unneeded values first and only then map the truthy ones e.g.

```
const channels = doctorMemberships.filter(m => {
    const patientId = m.members.find((id) => id !== doctor.id);
    return users.find((u) => u.id === patientId);
}).map(m => {
    const patientId = m.members.find((id) => id !== doctor.id);
    const patient = users.find((u) => u.id === patientId);

    return {
      id: m.channelId,
      name: patient.name,
      description: `Patient ID: ${patient.id}`,
      custom: {
        profileUrl: patient?.profileUrl,
      },
      updated: "",
      eTag: "",
    };
})
```

Or maybe even better to use reduce in order not to repeat similar codeblocks like this:
```
const channels = memberships.reduce((acc, m) => {
    const patientId = m.members.find((id) => id !== doctor.id);
    const patient = users.find((u) => u.id === patientId);
    if (!patient) return acc;
    return [...acc, {
      id: m.channelId,
      name: patient.name,
      description: `Patient ID: ${patient.id}`,
      custom: {
        profileUrl: patient?.profileUrl,
      },
      updated: "",
      eTag: "",
    }];
}, [])
```

Reply:
Nice catch. Honestly this whole code block is a bit redundant since we're using a fixed data set with one doctor that is connected with the two patients, and I don't expect any changes there any time soon, but I'll use the reduce solution